Zend/zend_operators.c

```diff
@@ -2132,6 +2132,57 @@ static int compare_double_to_string(double dval, zend_string *str) /* {{{ */
 }
 /* }}} */
 
+ZEND_API int ZEND_FASTCALL zend_equals_object(zval *op1, zval *op2, zend_uchar equals) /* {{{ */
+{
+	zval ret;
+	int retVal;
+	int result;
+	if (Z_TYPE_P(op1) == IS_OBJECT &&
+		Z_TYPE_P(op2) == IS_OBJECT &&
+		Z_OBJ_P(op1) == Z_OBJ_P(op2)) {
+		return 0;
+	} else if (Z_TYPE_P(op1) == IS_OBJECT) {
+		if (Z_OBJ_HANDLER_P(op1, do_operation)) {
+			result = Z_OBJ_HANDLER_P(op1, do_operation)(equals, &ret, op1, op2);
+			if (result == FAILURE) {
+				goto op2_equals_handler;
+			} else {
+				retVal = (Z_TYPE_INFO(ret) == IS_TRUE ? 0 : ZEND_UNCOMPARABLE);
+			}
+			return retVal;
+		} else {
+			if (Z_TYPE_P(op2) == IS_OBJECT) {
+				goto op2_equals_handler;
+			} else {
+				goto default_equals_handler;
+			}
+		}
+	} else if (Z_TYPE_P(op2) == IS_OBJECT) {
+op2_equals_handler:
+		if (Z_OBJ_HANDLER_P(op2, do_operation)) {
+			result = Z_OBJ_HANDLER_P(op2, do_operation)(equals, &ret, op1, op2);
+			if (result == FAILURE) {
+				goto default_equals_handler;
+			} else {
+				retVal = (Z_TYPE_INFO(ret) == IS_TRUE ? 0 : ZEND_UNCOMPARABLE);
+			}
+			return retVal;
+		} else {
+			if (Z_TYPE_P(op2) == IS_OBJECT) {
+				goto op2_equals_handler;
```

Comment:
Wouldn't this loop endlessly?

Reply:
Yep! That was the debug issue I was seeing. Unfortunately I couldn't build with debug due to ext/sockets having an issue.